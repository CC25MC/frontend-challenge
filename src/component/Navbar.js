import Container from "react-bootstrap/Container";
import Navbar from "react-bootstrap/Navbar";
import Dropdown from "react-bootstrap/Dropdown";
import { Fragment } from "react";

export const NavbarB = ({ title, filesError = [] }) => {
  return (
    <Navbar bg="dark" variant="dark">
      <Container>
        <Navbar.Brand>{title}</Navbar.Brand>
        {filesError.length > 0 && (
          <Fragment>
            <Navbar.Toggle />
            <Navbar.Collapse className="justify-content-end">
              <Dropdown>
                <Dropdown.Toggle variant="danger" id="dropdown-basic">
                  {filesError.length} Notifications
                </Dropdown.Toggle>

                <Dropdown.Menu>
                  {filesError.map((item, key) => (
                    <Dropdown.Item key={key}>
                      {item?.file} - {item?.error}
                    </Dropdown.Item>
                  ))}
                </Dropdown.Menu>
              </Dropdown>
            </Navbar.Collapse>
          </Fragment>
        )}
      </Container>
    </Navbar>
  );
};
