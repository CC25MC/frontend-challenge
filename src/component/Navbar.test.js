import { render } from "@testing-library/react";

import { NavbarB } from "./Navbar";

describe("Navbar tests", () => {
  const filesError = [
    {
      file: "test18.csv",
      error: "not found",
    },
  ];
  const title = "Test Challenge";

  it("Should render correctly", () => {
    const component = render(<NavbarB title={title} filesError={filesError} />);
    component.getByText(`${title}`);
    expect(component.container).toMatchSnapshot();
  });

  it("Should show the error notification", () => {
    const component = render(
      <NavbarB title={"Test Challenge"} filesError={filesError} />
    );
    component.getByText(`${filesError.length} Notifications`);
    expect(component.container).toMatchSnapshot();
  });
});
