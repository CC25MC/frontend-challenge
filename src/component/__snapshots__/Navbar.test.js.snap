// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests Should render correctly 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      class="container"
    >
      <span
        class="navbar-brand"
      >
        Test Challange
      </span>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="justify-content-end navbar-collapse collapse"
      >
        <div
          class="dropdown"
        >
          <button
            aria-expanded="false"
            class="dropdown-toggle btn btn-danger"
            id="dropdown-basic"
            type="button"
          >
            1
             Notifications
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Navbar tests Should show the error notification 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      class="container"
    >
      <span
        class="navbar-brand"
      >
        Test Challange
      </span>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="justify-content-end navbar-collapse collapse"
      >
        <div
          class="dropdown"
        >
          <button
            aria-expanded="false"
            class="dropdown-toggle btn btn-danger"
            id="dropdown-basic"
            type="button"
          >
            1
             Notifications
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
